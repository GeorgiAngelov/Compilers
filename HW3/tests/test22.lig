type a : {b1: b};
type b : {c1: c};
type c : {a1: a};

var x : a = {b1 = {c1 = {a1 = {b1 = nil}}}};
