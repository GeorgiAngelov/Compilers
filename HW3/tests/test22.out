(decls
   (decl-type:ok (id a) (type-struct (typed-id (id b1) (type-id b))))
   (decl-type:ok (id b) (type-struct (typed-id (id c1) (type-id c))))
   (decl-type:ok (id c) (type-struct (typed-id (id a1) (type-id a))))
   (decl-var:ok (id x) (type-id a) (struct-lit:{b1 : {c1 : {a1 : {b1 : nil}}}} (field-inits (field-init (id b1) (struct-lit:{c1 : {a1 : {b1 : nil}}} (field-inits (field-init (id c1) (struct-lit:{a1 : {b1 : nil}} (field-inits (field-init (id a1) (struct-lit:{b1 : nil} (field-inits (field-init (id b1) (nil:nil)))))))))))))))
Function main() not defined or has wrong type!
